### ISSUE TITLE
use of the extension protocol handler is failing on windows

### ISSUE THREAD

**Issue by IsaSih** [rancher-sandbox/rancher-desktop#4246]
### Actual Behavior

extensions-protocol.e2e.spec.ts:104:3 › Extensions protocol handler › build and install testing extension is failing on windows 11 by running the e2e tests via CI.

### Steps to Reproduce

Triggering e2e tests to run via CI on windows 11.

### Result

Test is showing the following error:
```
2) extensions-protocol.e2e.spec.ts:104:3 › Extensions protocol handler › build and install testing extension 

    Error: C:\actions-runner\_work\rancher-desktop\rancher-desktop\resources\win32\bin\nerdctl.exe exited with code 1

       at ..\pkg\rancher-desktop\utils\childProcess.ts:245

      243 |         message = `${ command } exited with signal ${ signal }`;
      244 |       }
    > 245 |       const error: SpawnError = new Error(message);
          |                                 ^
      246 |
      247 |       Object.defineProperties(error, {
      248 |         [ErrorCommand]: {

        at ChildProcess.<anonymous> (C:\actions-runner\_work\rancher-desktop\rancher-desktop\pkg\rancher-desktop\utils\childProcess.ts:245:33)
```

### Expected Behavior

The test should pass in all platforms

### Rancher Desktop Version

N/A

### Rancher Desktop K8s Version

N/A

### Which container engine are you using?

containerd (nerdctl)

### What operating system are you using?

Windows

### Operating System / Build Version

windows 11

### What CPU architecture are you using?

x64

### Linux only: what package format did you use to install Rancher Desktop?

None

**Comment by mook-as** [comment]
The test output:

```
  ok 21 credentials-server.e2e.spec.ts:630:5 › Credentials server › handles credHelpers › dcrd store uses credHelpers (1s)
  ok 22 extensions-protocol.e2e.spec.ts:65:3 › Extensions protocol handler › should load backend (8s)
  ok 23 extensions-protocol.e2e.spec.ts:69:3 › Extensions protocol handler › determine container engine in use (372ms)
Error: C:\actions-runner\_work\rancher-desktop\rancher-desktop\resources\win32\bin\rdctl.exe exited with code 1, retrying... (1/30)
  ok 24 extensions-protocol.e2e.spec.ts:77:3 › Extensions protocol handler › wait for buildkit (1s)
  -  25 extensions-protocol.e2e.spec.ts:85:3 › Extensions protocol handler › wait for docker context
  -  26 extensions-protocol.e2e.spec.ts:92:3 › Extensions protocol handler › wait for docker daemon to be up
Skipping packaging logs, not running in Cirrus CI
  x  27 extensions-protocol.e2e.spec.ts:104:3 › Extensions protocol handler › build and install testing extension (15s)
  -  28 extensions-protocol.e2e.spec.ts:111:3 › Extensions protocol handler › use extension protocol handler
```

The relevant logs seems to be:

    background.log  ┌2023-03-21T05:29:19.020Z: UnhandledRejectionWarning: Error: C:\…\resources\win32\bin\nerdctl.exe exited with code 1
    background.log  │    at ChildProcess.<anonymous> (C:\…\dist\app\background.js:45923:27)
    background.log  │    at ChildProcess.emit (node:events:527:28)
    background.log  │    at ChildProcess._handle.onexit (node:internal/child_process:291:12) {
    background.log  │  stdout: '',
    background.log  │  stderr: 'time="2023-03-21T05:29:19Z" level=fatal msg="cannot access containerd socket \\"/run/k3s/containerd/containerd.sock\\": no such file or directory"\n',
    background.log  │  code: 1,
    background.log  │  command: [
    background.log  │    'C:\\…\\resources\\win32\\bin\\nerdctl.exe',
    background.log  │    'namespace',
    background.log  │    'list',
    background.log  │    '--quiet'
    background.log  │  ]
    background.log  └}

So basically even though the UI shows things as ready, `nerdctl` was unable to talk to `containerd`.

**Comment by IsaSih** [comment]
Did not move it to done, because this test was skipped in the last e2e workflow run as docker deamon is failing to connect

**Comment by jandubois** [comment]
> Did not move it to done, because this test was skipped in the last e2e workflow run as docker deamon is failing to connect.

It is not clear to me what you expect to happen next. If you think the problem is not fixed, then the issue should be moved back to "Todo" and the person who worked on it (@ericpromislow) needs to be notified about it.

**Comment by ericpromislow** [comment]
I can't get it to fail in Windows (running `npm run test:e2e ... `; I don't have access to a Windows CI), but I did find another issue with the `extensions.e2e` test on non-Windows, created PR #4717 to deal with that.

**Comment by jandubois** [comment]
Closing and moving to Verify since the PR has already been merged
